test/e2e/framework/framework.go

```diff
@@ -123,9 +123,31 @@ func (f *Framework) BeforeEach() {
 
 	ginkgo.By("Creating kubernetes clients")
 
+	Expect(len(TestContext.KubeConfigs) > 0 && len(TestContext.KubeContexts) > 0).NotTo(BeTrue(),
+		"Mixing multiple KubeConfigs (contextless) and one KubeConfig variable"+
+			"with mutiple contexts is not supported.")
```

Comment:
So we've decided not to expose `KubeConfigs` as a CLI arg for now but perhaps we shouldn't make that assumption here. `ParseFlags` sets `TestContext.ClusterIDs = TestContext.KubeContexts` so it is expected that `KubeContexts` is set which will break these checks if `KubeConfigs` is set since it is expected that `len(ClusterIDs)` matches that of `KubeConfigs` below. 

Also all these checks and the interaction between the settings is a bit confusing. I think we can simplify by just making `KubeConfig` and `KubeConfigs` mutually exclusive, ie

```
if len(TestContext.KubeConfig) > 0 {
    Expect(len(TestContext.KubeConfigs)).To(BeZero(),
			"Either KubeConfig or KubeConfigs must be specified but not both")
    for _, context := range TestContext.KubeContexts {
        ...
    }
} else if len(TestContext.KubeConfigs) > 0 {
    Expect(len(TestContext.KubeConfig)).To(BeZero(),
			"Either KubeConfig or KubeConfigs must be specified but not both")
    Expect(len(TestContext.KubeConfigs)).To(Equal(len(TestContext.ClusterIDs)), "...")
    for _, kubeConfig := range TestContext.KubeConfigs {
        ....
    }
} else {
    Fail("One of KubeConfig or KubeConfigs must be specified")
}
```

Also, to simplify further,  I don't think we need both `KubeContexts` and `ClusterIDs` since the tests have been using the `KubeContexts` as cluster IDs anyway (and you've changed all other uses of `KubeContexts` to `ClusterIDs`) and you explicitly make then equal in `ParseFlags`.  So if using `KubeConfig` then `KubeContexts` and `ClusterIDs` are one and the same; if using `KubeConfigs` then `KubeContexts` isn't used - all that tells me that `KubeContexts` isn't needed.

Reply:
On the check, I agree, I like your proposal better.

On the difference between ClusterIDs and contexts, I disagree, our E2E tests were assuming that clusterIDs == contexts, which is important specially in the e2e tests which check clusters, but also important for printing.

Reusing the E2E tests from subctl needs to dissociate those two things which are logically different, just happen to be the same in our environment.

May be I should move the ClusterID array population to the BeforeEach part, explaining the logic.

Reply:
> May be I should move the ClusterID array population to the BeforeEach part, explaining the logic.

yeah - I was thinking that too, ie assign `ClusterIDs` to `KubeContexts` only if it's nil/empty.